gpkit/constraints/set.py

```diff
@@ -177,9 +180,9 @@ def as_gpconstr(self, x0):
         """Returns GPConstraint approximating this constraint at x0
 
         When x0 is none, may return a default guess."""
-        cs = ConstraintSet([constr.as_gpconstr(x0) for constr in self])
-        cs.substitutions.update(self.substitutions)
-        return cs
+        gpconstrs = [constr.as_gpconstr(x0) for constr in self]
+        return ConstraintSet(gpconstrs,
+                             self.substitutions, recursesubs=False)
```

Comment:
making sure I understand the logic: we can set `recursesubs=False` because we assume that the substitutions of any submodels do not change between SP iterations (?). What about the first run through? Is that handled by solve methods?

Or does this assume that the user does not change substitutions at all?